api/token/service.go

```diff
@@ -0,0 +1,78 @@
+package token
+
+import (
+	"context"
+	"errors"
+
+	"github.com/shellhub-io/shellhub/api/store"
+	"github.com/shellhub-io/shellhub/pkg/models"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+var ErrUnauthorized = errors.New("unauthorized")
+var ErrUserNotFound = errors.New("user not found")
+var ErrNamespaceNotFound = errors.New("namespace not found")
+var ErrDuplicateID = errors.New("user already member of this namespace")
+var ErrUserOwner = errors.New("cannot remove this user")
+var ErrTokenNotFound = errors.New("token not found for this namespace")
+
+type Service interface {
+	ListTokens(ctx context.Context, namespace string) ([]models.Token, error)
+	CreateToken(ctx context.Context, namespace string) (*models.Token, error)
+	GetToken(ctx context.Context, namespace string) (*models.Token, error)
+	DeleteToken(ctx context.Context, namespace string) error
+	SetReadOnly(ctx context.Context, name string) error
+}
+
+type service struct {
+	store store.Store
+}
+
+func NewService(store store.Store) Service {
+	return &service{store}
+}
+
+func (s *service) ListTokens(ctx context.Context, namespace string) ([]models.Token, error) {
+	ns, _ := s.store.GetNamespace(ctx, namespace)
+	if ns == nil {
+		return []models.Token{}, ErrTokenNotFound
+	}
+
+	tokens := []models.Token{}
+	for _, token := range ns.Token.(primitive.A) {
```

Comment:
Avoid using `primitive` as we ought to use the `store` to keep the storage use isolated.